core/src/nonce.rs

```diff
@@ -37,3 +46,96 @@ where
         self.0.as_iter()
     }
 }
+
+// To distinguish between legacy nonces and expirable nonces
+// we use a specific prefix EXPIRABLE_NONCE_PREFIX. Expirable nonces
+// have the following structure: [word_position, bit_position].
+// Where word_position = [ EXPIRABLE_NONCE_PREFIX , <8 bytes timestamp in ms>, <20 random bytes> ]
+pub struct ExpirableNonce {
+    pub prefix: [u8; 4],
+    pub timestamp: u64,
```

Comment:
Let's use already built abstraction:
```suggestion
    pub deadline: Deadline,
```